### ISSUE TITLE
issue with select plugin

### ISSUE THREAD

**Issue by Aenzenith** [DataTables/Vue#9]
Today when i run npm update i realized table component is not working anymore. I searched source of issue and it was related to new version of datatables.net-select plugin. The plugin runs perfectly with 1.4.0 but i im getting error like this when i use 1.5.0
`Uncaught (in promise) TypeError: Cannot set properties of undefined (setting '$')`

Let you know. For now i downgraded datatables.net-select to 1.4.0 and issue is fixed.

**Comment by AllanJard** [comment]
See [the information in this blog post](https://datatables.net/blog/2022-11-10-esm#Backwards-compatibility).

**Comment by Aenzenith** [comment]
Thanks for answer,

`If you do that with the ES module, you will get an error such as "Connot set properties of undefined (setting '$')". The fix is to not execute the imported module as a function - e.g. just do:
`

`import DataTable from 'datatables.net';`


That was exactly what i did. I'm including like that way.


```
...

import { ref, onMounted, onUnmounted, defineEmits, defineProps, inject } from 'vue';
import { can } from '@/Utils/Permission.js';
import DataTable from 'datatables.net-vue3';
import Select from 'datatables.net-select';

DataTable.use(Select);
let dt;
const table = ref();

onMounted(function () {
    if (can('view')) {
        dt = table.value.dt();
    }
    window.addEventListener('click', selected);
});


...

```

If there is a thing that i am missing can you explain please?

**Comment by AllanJard** [comment]
Sorry - this was an incompatibility in the Vue component for DataTables and DataTables 1.13 (due to the new ES module support in 1.13).

I've made two releases to fix this:

1. v1.0.1 of this plug-in has been release which uses the same API as before, but has a fixed dependency of DataTables 1.12.1. The v1.x releases will not work with newer releases of DataTables!
2. v2.0.0 has also been released with an updated API that supports DataTables 1.13 and newer.

The new API is very similar to what we had before, but instead of assigning extensions through the `.use()` method, you now build up the DataTables module you want and assign it to the component - e.g. in this case:

```js
import DataTable from 'datatables.net-vue3';
import DataTablesLib from 'datatables.net';
import 'datatables.net-select';

DataTable.use(DataTablesLib);
```

Everything else remains the same.
